pkg/statistics/histogram.go

```diff
@@ -1153,55 +1193,58 @@ func (hg *Histogram) OutOfRangeRowCount(
 			// Calculate the percentage of "the shaded area" on the right side.
 			rightPercent = (math.Pow(boundR-actualL, 2) - math.Pow(boundR-actualR, 2)) / math.Pow(histWidth, 2)
 		}
+		// Recalculate the out-of-range percentage to account for time decay (stale histogram).
+		// We do this by moving the predicate range closer to the histogram range, and
+		// recalculating the percentage of the shaded area.
+		actualL = l
+		actualR = r
+		// Predicate entirely on the right side of the histogram envelope.
+		if actualL > histR {
+			entirelyOutOfRange = true
+			adjLeft := histR
+			adjRight := adjLeft + predWidth
+			timeAdjRight = (math.Pow(boundR-adjLeft, 2) - math.Pow(boundR-adjRight, 2)) / math.Pow(histWidth, 2)
```

Comment:
How to understand this part, could you add more comments to explain this part? 🙏